sdks/python/apache_beam/transforms/core.py

```diff
@@ -189,6 +276,54 @@ def finish_bundle(self):
     """
     pass
 
+  def new_tracker(self, restriction):
+    """Produces a new ``RestrictionTracker`` for the given restriction.
+
+    Args:
+      restriction: an object that defines a restriction as identified by the
+        current ``DoFn``. For example, a tuple that gives a range of positions
+        for a ``DoFn`` that reads files based on byte positions.
+    Returns: an object of type ``RestrictionTracker``.
+    """
+    pass
+
+  def restriction_coder(self):
+    """Returns a ``Coder`` for restrictions produced by the current ``DoFn``.
+
+    Returns:
+      an object of type ``Coder``.
+    """
+    return coders.registry.get_coder(object)
+
+  def initial_restriction(self, element):
+    """Produces an initial restriction for the given element."""
+    pass
+
+  def split(self, element, restriction):
+    """Splits the given element and restriction.
+
+    Returns an iterator of restrictions. The total set of elements produced by
+    reading input element for each of the returned restrictions should be the
+    same as the total set of elements produced by reading the input element for
+    the input restriction.
+
+    TODO(chamikara): give suitable hints for performing splitting, for example
+    number of parts or size in bytes.
+    """
+    yield restriction
+
+  @staticmethod
+  def resume(resume_delay=0):
```

Comment:
Is this intended to be used as 

    return self.resume(...)

Maybe it'd be better to put this as a static method on ProcessContinuation.